asset/src/kafka_sender/schema.ts

```diff
@@ -1,14 +1,47 @@
-import { ConvictSchema } from '@terascope/job-components';
+import {
+    ConvictSchema, ValidatedJobConfig, getOpConfig, get
+} from '@terascope/job-components';
 import { KafkaSenderConfig } from './interfaces';
 
+function fetchConfig(job: ValidatedJobConfig) {
+    const opConfig = getOpConfig(job, 'kafka_sender');
+    if (opConfig == null) throw new Error('Could not find elasticsearch_bulk operation in jobConfig');
+    return opConfig as KafkaSenderConfig;
+}
+
 export default class Schema extends ConvictSchema<KafkaSenderConfig> {
+    validateJob(job: ValidatedJobConfig) {
+        const opConfig = fetchConfig(job);
+        const kafkaConnectors = get(this.context, 'sysconfig.terafoundation.connectors.kafka');
+        if (kafkaConnectors == null) throw new Error('Could not find kafka connector in terafoundation config');
+
+        // check to verify if connection map provided is
+        // consistent with sysconfig.terafoundation.connectors
+        if (opConfig.connection_map) {
+            for (const [, value] of Object.entries(opConfig.connection_map)) {
+                if (!kafkaConnectors[value]) {
+                    throw new Error(`A connection for [${value}] was set on the elasticsearch_bulk connection_map but is not found in the system configuration [terafoundation.connectors.elasticsearch]`);
```

Comment:
If you are going to copy and paste, make sure to fix references

```suggestion
                    throw new Error(`A connection for [${value}] was set on the kafka_sender connection_map but is not found in the system configuration [terafoundation.connectors.elasticsearch]`);
```